### DISCUSSION TITLE
Restricting specific dependency updates to patch

### DISCUSSION THREAD

**Discussion by dkrowl** [renovatebot/renovate#18732] [Request Help]
I have packageRules that allow minor and patch updates to be automerged. But I want to restrict specific dependencies to patch-versions only.

I have tries this:

````json
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "automerge": true
    },
    {
       "matchPackagePatterns": ["^com.hazelcast"],
       "matchUpdateTypes": ["patch"]
    }]
```` 

Apparently this isn't a correct or even a working way to do this. I also tried allowedVersions but it seems that the first matchUpdateTypes still is valid with the hazelcast-dependencies.

**Comment by viceice** [comment]
one way to solve:
```json
"packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "automerge": true
    },
    {
       "matchPackagePatterns": ["^com.hazelcast"],
       "matchUpdateTypes": ["minor"],
       "automerge": false
    },
    {
       "matchPackagePatterns": ["^com.hazelcast"],
       "separateMinorPatch": true
    }
]
```

**Reply by dkrowl** [reply]
@viceice 
but then the dependencies aren't updated to a new patch-version if a later minor-version is available. Rgiht?

**Reply by viceice** [reply]
yes, see https://docs.renovatebot.com/configuration-options/#separateminorpatch to get them 😉

**Reply by dkrowl** [reply]
but then the patch automerge won't work.. oh, man.

**Reply by viceice** [reply]
you need a second rule for separate`, because it's applied before the update typoe is known

```json
"packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "automerge": true
    },
    {
       "matchPackagePatterns": ["^com.hazelcast"],
       "matchUpdateTypes": ["minor"],
       "automerge": false
    },
    {
       "matchPackagePatterns": ["^com.hazelcast"],
       "separateMinorPatch": true
    }
]
```
